processing/src/main/java/io/druid/query/lookup/LookupReferencesManager.java

```diff
@@ -137,7 +176,11 @@ public boolean remove(String lookupName)
   {
     final LookupExtractorFactory lookupExtractorFactory = lookupMap.remove(lookupName);
     if (lookupExtractorFactory != null) {
-      LOGGER.debug("Removing lookup [%s]", lookupName);
+      LOGGER.debug("Removed lookup [%s]", lookupName);
+      if (!isClosed() && lookupSnapshotTaker != null) {
+        // if it is closed the final snapshot will be taken at stop() method
+        lookupSnapshotTaker.takeSnapshot(getAllAsList());
```

Comment:
i am not sure i get this.  For all the put and remove i am asserting that is started and the block is synchronized around lock.
 ```  synchronized (lock) {
      assertStarted();
```